zstacklib/zstacklib/utils/libvirt_singleton.py

```
@@ -8,8 +8,11 @@
 
 @singleton.singleton
 class LibvirtSingleton(object):
-    def __init__(self):
-        self.conn = libvirt.open('qemu:///system')
+    def __init__(self, uri='qemu:///system'):
+        self.conn = libvirt.open(uri)
+        if not self.conn:
+            raise Exception('Failed to open connection to the hypervisor')
```

Ruff (0.15.2)

[warning] Create your own exception

(TRY002)